iocore/fs/FileChange.h

```diff
@@ -0,0 +1,84 @@
+/** @file FileChange.h
+
+  Watch for file system changes.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+#include "tscore/ink_config.h"
+
+#include <thread>
+#include <chrono>
+#include <set>
+#include <unordered_map>
+#include <vector>
+#include <shared_mutex>
+#include "tscore/ts_file.h"
+#include "P_EventSystem.h"
+
+#if TS_USE_INOTIFY
+#include <sys/inotify.h>
+#else
+// implement this
+#endif
+
+using watch_handle_t = int;
+
+// File watch info
+struct file_info {
+  ts::file::path path;
+  Continuation *contp;
+};
+
+constexpr int MAX_WATCHES = 10000;
```

Comment:
I can't seem to find anywhere that uses this? Also curious is this just an arbitrary number?

Reply:
I forgot to remove this from back when I tried using a separate fd for each watch.  Removed.